Plugins/Creature/Creature.cpp

```diff
@@ -1047,12 +1047,26 @@ NWNX_EXPORT ArgumentStack SetClassByPosition(ArgumentStack&& args)
     {
         const auto position = args.extract<int32_t>();
         const auto classID = args.extract<int32_t>();
+        const auto bUpdateLevels = args.extract<int32_t>();
+        const auto classIDold = pCreature->m_pStats->GetClass(position);
           ASSERT_OR_THROW(position >= 0);
           ASSERT_OR_THROW(position <= 2);
           ASSERT_OR_THROW(classID >= Constants::ClassType::MIN);
           ASSERT_OR_THROW(classID <= Constants::ClassType::MAX);
 
         pCreature->m_pStats->SetClass(position, classID);
+
+        if (bUpdateLevels)
+        {
+            auto pLevelStats = pCreature->m_pStats->m_lstLevelStats;
```

Comment:
This should be `auto&`, you don't want to make a copy of `m_lstLevelStats`. `auto x = y;` makes a copy of `y` and puts it in `x`. `auto& x = y` makes `x` an alias of `y` (simplified, but close enough).

Reply:
Also, nit, don't use the `p` prefix unless it's actually a pointer. just `levelStats` or even `levels` would do here as a name.